src/particles/particle_creation.hpp

```diff
@@ -151,6 +157,298 @@ template <ParticleType particleType> struct ParticleCreationTraits {
 	}
 };
 
+#if AMREX_SPACEDIM == 3
+// Specialization for StochasticStellarPop particles
+template <> struct ParticleCreationTraits<ParticleType::StochasticStellarPop> {
+	// Specialized nested ParticleChecker for StochasticStellarPop particles
+
+	static constexpr amrex::Real eps_star = 0.5; // fraction of gas mass that goes into star particles
+	static constexpr amrex::Real eps_ff = 0.5;   // efficiency per free fall time
+	static constexpr amrex::Real J = 0.5;	     // Jeans parameter
+
+	// Constants for the Chabrier IMF
+	static constexpr amrex::Real m_star_high = 8.0 * C::M_solar; // all stars above this mass are considered high mass stars
+	static constexpr amrex::Real m_imf_min = 0.08 * C::M_solar;  // lower limit of the IMF
+	static constexpr amrex::Real m_imf_max = 120.0 * C::M_solar; // high mass limit of the IMF
+	static constexpr amrex::Real m_imf_break = 1.0 * C::M_solar; // IMF is lognormal below this mass and powerlaw above
+	static constexpr amrex::Real imf_disp = 0.55;		     // dispersion of the lognormal IMF
+	static constexpr amrex::Real imf_mu = 32.599;		     //=log10(0.2 * C::M_solar), mean of the lognormal IMF, avoid compiler error
+	static constexpr amrex::Real alpha = 2.35;		     // slope of the powerlaw
+
+	// Write out expression because compiler error for nested gcem
```

Comment:
I think it is super-confusing to have half the calculation done here this way, which is sort of halfway between writing out the answer and doing the calculation all on the fly with constexpr's. I suggest just writing out the values of fstar_high and m_star_high_avg explicitly, with comments giving the algebraic expressions from which they are derived? It would make the code shorter and much more readable.